hr_payroll_analysis/models/hr_payslip.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding:utf-8 -*-
+from openerp import models, fields, api
+
+
+class HrPayslip(models.Model):
+    _inherit = 'hr.payslip'
+    
+    analysis_line_ids = fields.One2many('hr.payslip.analysis.line', 
+                                        'payslip_id', 'Analysis Lines')
+
+    @api.multi
+    def process_sheet(self):
+        """
+        Create the analysis lines when the payslip is confirmed
+        """
+        super(HrPayslip, self).process_sheet()
+        self.remove_analysis_lines()
+        self.compute_analysis_lines()
+    
+    @api.multi
+    def remove_analysis_lines(self):
+        """
+        Make sure no analysis line has already been computed
+        """
+        for payslip in self:
+            for line in payslip.analysis_line_ids:
+                line.unlink()
+    
+    @api.multi
+    def compute_analysis_lines(self):
+
+        analysis_line_obj = self.env['hr.payslip.analysis.line']
+
+        for payslip in self:
+
+            required_lines = [
+                line for line in payslip.details_by_salary_rule_category
+                if line.salary_rule_id.include_in_payroll_analysis and
+                line.total
```

Comment:
you may use payslip.details_by_salary_rule_category.filtered(lambda line: line.salary_rule_id.include_in_payroll_analysis and line.total)
which returns a record set